cns/kubecontroller/nodenetworkconfig/conversion.go

```diff
@@ -92,6 +92,15 @@ func CreateNCRequestFromStaticNC(nc v1alpha.NetworkContainer) (*cns.CreateNetwor
 		PrefixLength: uint8(subnetPrefix.Bits()),
 	}
 
+	if nc.Type == v1alpha.VNETBlock {
+		nodeIP, nerr := netip.ParsePrefix(nc.NodeIP)
+		if nerr != nil {
+			return nil, errors.Wrapf(err, "IP: %s", nc.NodeIP)
+		}
+		subnet.IPAddress = nodeIP.Addr().String()
```

Comment:
using node IP here kind of breaks the premise of podsubnet that you can have different policies for pods and nodes, I think. If you SNAT some traffic against the node IP, then rules you put on the podsubnet would not get applied to that traffic vs if you SNAT against an IP in the pod subnet. Do we want to make this change in behavior?

Reply:
For example if a customer has a NSG rule on the podsubnet to block traffic to some peered vnet, the podsubnet traffic would get SNAT'ed with Node IP here and the traffic would reach. So customer thinks they've blocked the traffic from pods to peered vnet, but the traffic would still work.

Reply:
afaik in podsubnet, we snat only if traffic is destined to azure dns inside guest. so it won't be a problem right?

Reply:
@tamilmani1989 : So does the peering traffic flow directly through NMAgent route tables on the Vnet backbone without SNAT in case of peering?

Reply:
Yes I was mixing up overlay and podsubnet snat scenarios, in podsubnet it is only snatted with primary IP of NC for azure dns traffic, I can't imagine that customers would want different DNS policies on node and pod subnets, but I think it's still worth calling out in AKS release notes as a breaking change

Reply:
Sounds good. Will add it to the release notes for CNS